cpp/src/arrow/array/data.h

```diff
@@ -434,6 +435,28 @@ struct ARROW_EXPORT ArraySpan {
     return GetValues<T>(i, this->offset);
   }
 
+  /// \brief Access a buffer's data as a span
+  ///
+  /// \param i The buffer index
+  /// \param length The required length (in number of typed values) of the requested span
+  /// \pre i > 0
+  /// \pre length <= the length of the buffer (in number of values) that's expected for
+  /// this array type \return A span<const T> of the requested length
+  template <typename T>
+  util::span<const T> GetSpan(int i, int64_t length) const {
```

Comment:
Copy and paste the docstring here (change `span<const T>` to `span<T>`) -- these show up in auto-complete pop-ups on IDEs and editors, so we have to be repetitive.